server/controllers/finance/reports/collection_capacity/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @overview ./finance/reports/collection_capacity/
+ */
+
+const _ = require('lodash');
+const moment = require('moment');
+
+const ReportManager = require('../../../../lib/ReportManager');
+const db = require('../../../../lib/db');
+
+module.exports.report = report;
+
+// path to the template to render
+const TEMPLATE = './server/controllers/finance/reports/collection_capacity/report.handlebars';
+
+const DEFAULT_OPTIONS = {
+  orientation : 'portrait',
+  footerRight : '[page] / [toPage]',
+  footerFontSize : '7',
+};
+
+/**
+ * @method report
+ *
+ * @description
+ * The HTTP interface which actually creates the report.
+ */
+async function report(req, res, next) {
+  try {
+    const qs = _.extend(req.query, DEFAULT_OPTIONS);
+    const {
+      dateFrom, dateTo, accountId, cashboxId,
+    } = req.query;
+    const metadata = _.clone(req.session);
+
+    const rpt = new ReportManager(TEMPLATE, metadata, qs);
+
+    const CASH_PAYMENT_TRANSACTION_TYPE = 2;
+    const CAUTION_TRANSACTION_TYPE = 19;
+
+    const patients = `
+      SELECT DATE(p.registration_date) AS registration_date, COUNT(*) registrations
+      FROM patient p 
+      WHERE (DATE(p.registration_date) BETWEEN ? AND ?)
+        AND project_id = ?
+      GROUP BY DATE(p.registration_date)
+    `;
+
+    const invoices = `
+      SELECT DATE(i.date) AS invoice_date, SUM(i.cost) total_invoiced
+      FROM invoice i 
+      WHERE DATE(i.date) BETWEEN ? AND ?
+        AND reversed = 0 AND project_id = ?
+      GROUP BY DATE(i.date)
+    `;
+
+    const payments = `
+      SELECT DATE(gl.trans_date) trans_date, SUM(IFNULL(debit_equiv - credit_equiv, 0)) AS total_paid 
+      FROM general_ledger gl
+      JOIN cash c ON c.uuid = gl.record_uuid
+      WHERE gl.transaction_type_id IN (${CASH_PAYMENT_TRANSACTION_TYPE}, ${CAUTION_TRANSACTION_TYPE})
+        AND (DATE(gl.trans_date) >= ? AND DATE(gl.trans_date) <= ?)
+        AND gl.account_id = ? AND c.reversed = 0 
+      GROUP BY DATE(gl.trans_date)
+    `;
+
+    const query = `
+      SELECT 
+        i.invoice_date, pa.registrations, i.total_invoiced, IFNULL(p.total_paid, 0) total_paid,
+        IF(pa.registrations <> 0, (IFNULL(i.total_invoiced, 0) / pa.registrations), 0) avg_cost,
+        IF(i.total_invoiced <> 0, (IFNULL(p.total_paid, 0) / i.total_invoiced), 0) recovery_capacity
+      FROM (${invoices}) i
+      LEFT JOIN (${payments}) p ON p.trans_date = i.invoice_date
+      LEFT JOIN (${patients}) pa ON pa.registration_date = i.invoice_date
```

Comment:
If I'm reading this right ... this will only show rows for which an invoice exists, correct?  So if we had a day with patients and cash payments, but no invoices, it won't show that day?